
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="geneXplain platform API tutorial 2022">
      
      
      
        <meta name="author" content="pstegma">
      
      <link rel="shortcut icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.0.2">
    
    
      
        <title>Platform tools and parameters - Tutorial for geneXplain platform APIs 2022</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.38780c08.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.3f72e892.min.css">
        
          
          
          <meta name="theme-color" content="#009485">
        
      
    
    
    
      
        
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="teal" data-md-color-accent="">
      
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#platform-tools-and-parameters" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="." title="Tutorial for geneXplain platform APIs 2022" class="md-header-nav__button md-logo" aria-label="Tutorial for geneXplain platform APIs 2022">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            Tutorial for geneXplain platform APIs 2022
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Platform tools and parameters
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/genexplain/genexplain-api/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="Tutorial for geneXplain platform APIs 2022" class="md-nav__button md-logo" aria-label="Tutorial for geneXplain platform APIs 2022">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Tutorial for geneXplain platform APIs 2022
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/genexplain/genexplain-api/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="index.html" class="md-nav__link">
      Introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="first_example.html" class="md-nav__link">
      A first example&#58; Enrichment of gene functions in a gene set
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Platform tools and parameters
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="basics.html" class="md-nav__link md-nav__link--active">
      Platform tools and parameters
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#listing-tools-and-parameters-using-genexplainr" class="md-nav__link">
    Listing tools and parameters using geneXplainR
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#listing-tools-and-parameters-using-the-commandline-program-apps" class="md-nav__link">
    Listing tools and parameters using the commandline program apps
  </a>
  
    <nav class="md-nav" aria-label="Listing tools and parameters using the commandline program apps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#output-from-apps-including-parameters" class="md-nav__link">
    Output from apps including parameters
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fetching-parameter-information-for-specific-tools-using-the-parameters-utility" class="md-nav__link">
    Fetching parameter information for specific tools using the parameters utility
  </a>
  
    <nav class="md-nav" aria-label="Fetching parameter information for specific tools using the parameters utility">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#output-produced-by-parameters-for-platform-tools" class="md-nav__link">
    Output produced by parameters for platform tools
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="api_overview.html" class="md-nav__link">
      Overview of APIs
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Example workflow&#58; Analysis of ChIP-seq data
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Example workflow&#58; Analysis of ChIP-seq data" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        <span class="md-nav__icon md-icon"></span>
        Example workflow&#58; Analysis of ChIP-seq data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="chipseq_analysis_intro.html" class="md-nav__link">
      Analysis overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="chipseq_analysis_r.html" class="md-nav__link">
      geneXplainR
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="chipseq_analysis_java.html" class="md-nav__link">
      Java program
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="chipseq_analysis_json.html" class="md-nav__link">
      JSON interface
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#listing-tools-and-parameters-using-genexplainr" class="md-nav__link">
    Listing tools and parameters using geneXplainR
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#listing-tools-and-parameters-using-the-commandline-program-apps" class="md-nav__link">
    Listing tools and parameters using the commandline program apps
  </a>
  
    <nav class="md-nav" aria-label="Listing tools and parameters using the commandline program apps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#output-from-apps-including-parameters" class="md-nav__link">
    Output from apps including parameters
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fetching-parameter-information-for-specific-tools-using-the-parameters-utility" class="md-nav__link">
    Fetching parameter information for specific tools using the parameters utility
  </a>
  
    <nav class="md-nav" aria-label="Fetching parameter information for specific tools using the parameters utility">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#output-produced-by-parameters-for-platform-tools" class="md-nav__link">
    Output produced by parameters for platform tools
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/genexplain/genexplain-api/edit/master/docs/basics.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="platform-tools-and-parameters">Platform tools and parameters</h1>
<p>There are three main types of platform applications</p>
<ol>
<li>Analysis tools that cover functionality like data preparation, conversion, or data analysis</li>
<li>Importers that import certain data types into the platform workspace</li>
<li>Exporters that export data from the platform workspace</li>
</ol>
<p>The programming and commandline interfaces provide functions to list available analysis tools,
importers and exporters for a given platform server and user account and furthermore to
retrieve information about their input options/parameters.</p>
<p>In the following, we focus on geneXplainR and commandline utilities provided by the
genexplain-api package. Corresponding methods of the Java API are listed
<a href="api_overview.html#main-api-methods">here</a>.</p>
<p>We recommend to use genexplain-api commandline tools for tool listings and parameters
also when working mainly with geneXplainR.</p>
<h2 id="listing-tools-and-parameters-using-genexplainr">Listing tools and parameters using geneXplainR</h2>
<p>The example code below shows the geneXplainR functions to
list available applications and to get information about parameters
for a platform tool, <code>gx.analysis.list</code> and <code>gx.analysis.parameters</code>.</p>
<p>The output of <code>gx.analysis.list</code> is a table containing tool names in the <em>Name</em>
column and a column with the program group which corresponds to the name
of the analysis tool folder in which a tool can be found in the graphical
web interface.</p>
<p>The <code>gx.analysis.parameters</code> output, shown below for <em>Annotate table</em>, is
a table with parameter names as row ids and a column with short descriptions.</p>
<p>For importers and exporters, corresponding functionality is implemented by
<code>gx.importers</code> and <code>gx.exporters</code> which list available import and export functions,
respectively, and by <code>gx.import.parameters</code> as well as <code>gx.export.parameters</code> which
extract parameter information for specified importers and exporters,
respectively.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code><span class="o">&gt;</span> <span class="nf">library</span><span class="p">(</span><span class="n">geneXplainR</span><span class="p">)</span>

<span class="c1"># Server, username and password need to be replaced with valid input values.</span>
<span class="c1">#</span>
<span class="o">&gt;</span> <span class="nf">gx.login</span><span class="p">(</span><span class="s">&quot;https://platform.genexplain.com&quot;</span><span class="p">,</span> <span class="s">&quot;someuser@email.io&quot;</span><span class="p">,</span> <span class="s">&quot;12345&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="nf">gx.analysis.list</span><span class="p">()</span>
              <span class="n">Group</span>                              <span class="n">Name</span>
<span class="m">1</span> <span class="n">Data</span> <span class="n">manipulation</span>                  <span class="n">Annotate</span> <span class="n">diagram</span>
<span class="m">2</span> <span class="n">Data</span> <span class="n">manipulation</span>                    <span class="n">Annotate</span> <span class="n">table</span>
<span class="m">3</span> <span class="n">Data</span> <span class="n">manipulation</span>         <span class="n">Annotate</span> <span class="n">track</span> <span class="n">with</span> <span class="n">genes</span>
<span class="m">4</span> <span class="n">Data</span> <span class="n">manipulation</span>               <span class="n">CR</span> <span class="n">cluster</span> <span class="n">selector</span>
<span class="m">5</span> <span class="n">Data</span> <span class="n">manipulation</span> <span class="n">Calculate</span> <span class="n">weighted</span> <span class="n">mutation</span> <span class="n">score</span>
<span class="m">6</span> <span class="n">Data</span> <span class="n">manipulation</span>                      <span class="n">Check</span> <span class="n">quotas</span>
<span class="kc">...</span>

<span class="o">&gt;</span> <span class="nf">gx.analysis.parameters</span><span class="p">(</span><span class="s">&quot;Annotate table&quot;</span><span class="p">)</span>
                                                                                                                         <span class="n">description</span>
<span class="n">inputTablePath</span>                                                                                    <span class="n">Table</span> <span class="n">with</span> <span class="nf">experimental</span><span class="p">(</span><span class="n">test</span><span class="p">)</span> <span class="n">data</span>
<span class="n">species</span>                                                                           <span class="n">Species</span> <span class="n">to</span> <span class="n">be</span> <span class="n">used</span> <span class="n">during</span> <span class="nf">matching </span><span class="p">(</span><span class="n">if</span> <span class="n">applicable</span><span class="p">)</span>
<span class="n">annotationCollectionPath</span>                                                                            <span class="n">Data</span> <span class="n">collection</span> <span class="n">with</span> <span class="n">annotations</span>
<span class="n">annotationColumns</span>                                                                                       <span class="n">Names</span> <span class="n">for</span> <span class="n">annotation</span> <span class="n">columns</span>
<span class="n">replaceDuplicates</span>        <span class="n">If</span> <span class="n">input</span> <span class="n">table</span> <span class="n">was</span> <span class="n">already</span> <span class="n">annotated</span> <span class="n">by</span> <span class="n">this</span> <span class="n">source</span><span class="p">,</span> <span class="n">old</span> <span class="n">annotation</span> <span class="n">columns</span> <span class="n">will</span> <span class="n">be</span> <span class="n">removed</span> <span class="n">from</span> <span class="n">the</span> <span class="n">result</span>
<span class="n">outputTablePath</span>                                                                                                         <span class="n">Output</span> <span class="n">table</span>
</code></pre></div>
</td></tr></table>

<h2 id="listing-tools-and-parameters-using-the-commandline-program-apps">Listing tools and parameters using the commandline program <em>apps</em></h2>
<p>The commandline program <a href="api_overview.html#apps-listing-available-tools">apps</a> can retrieve
the list of available platform tools as well as an extended table that additionally contains
information about tool parameters.</p>
<p>For a simple listing of tools the input JSON file may look as follows, where <code>server</code>, <code>user</code>
and <code>password</code> properties need to be replaced with valid input values.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code><span class="p">{</span>
    <span class="nt">&quot;server&quot;</span><span class="p">:</span>          <span class="s2">&quot;https://platform.genexplain.com&quot;</span><span class="p">,</span>
    <span class="nt">&quot;user&quot;</span><span class="p">:</span>            <span class="s2">&quot;someuser@email.io&quot;</span><span class="p">,</span>
    <span class="nt">&quot;password&quot;</span><span class="p">:</span>        <span class="s2">&quot;12345&quot;</span><span class="p">,</span>
    <span class="nt">&quot;withParameters&quot;</span><span class="p">:</span>  <span class="kc">false</span><span class="p">,</span>
    <span class="nt">&quot;withGalaxy&quot;</span><span class="p">:</span>      <span class="kc">false</span>
<span class="p">}</span>
</code></pre></div>
</td></tr></table>

<p>The corresponding shell command is as follows, assuming the JSON object above
is stored in a file named <em>simple_tool_listing.json</em>.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code>java -jar genexplain-api.jar apps simple_tool_listing.json
</code></pre></div>
</td></tr></table>

<p>The output consists of a list of analysis tools preceded by the program group to which a tool
is assigned, e.g. <em>Statistical analysis</em>. The group name corresponds to the
analysis tool folder in the graphical web interface where a tool can be found.
Please note that in function calls only the tool name without the program group name is required.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code><span class="err">Data manipulation/Annotate diagram</span>
<span class="err">Data manipulation/Annotate table</span>
<span class="err">Data manipulation/Annotate track with genes</span>
<span class="err">Data manipulation/CR cluster selector</span>
<span class="err">Data manipulation/Calculate weighted mutation score</span>
<span class="err">Data manipulation/Check quotas</span>
<span class="err">Data manipulation/Composite module to proteins</span>
<span class="err">Data manipulation/Convert table</span>
<span class="err">Data manipulation/Convert table to VCF track</span>
<span class="err">...</span>
<span class="err">Statistical analysis/Train random forest</span>
<span class="err">Statistical analysis/Up and Down Identification</span>
<span class="err">Statistical analysis/Variance filter</span>
<span class="err">Statistical analysis/t-SNE</span>
<span class="err">miRNA analysis/Analyze miRNA target enrichment</span>
<span class="err">miRNA analysis/Create miRNA promoters</span>
<span class="err">miRNA analysis/Get miRNA targets</span>
<span class="err">miRNA analysis/miRNA feed forward loops</span>
</code></pre></div>
</td></tr></table>

<p>The output can be extended to <a href="https://usegalaxy.org">Galaxy</a> tools integrated
in the specified platform instance by setting the <code>withGalaxy</code> property to <code>true</code>.
Furthermore, the output can include information about tool parameters by setting
the <code>withParameters</code> property to <code>true</code>.</p>
<p>An example output with parameters is shown <a href="basics.html#output-from-apps-including-parameters">below</a>.
The output is a table with the following columns.</p>
<table>
<thead>
<tr>
<th>Column name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Tool folder/name</td>
<td>String consisting of program group and tool name separated by <strong>/</strong>. The program group corresponds to the graphical interface folder where the tool can be found.</td>
</tr>
<tr>
<td>API name</td>
<td>The tool name that needs to be specified in API calls, e.g. gx.analysis of geneXplainR</td>
</tr>
<tr>
<td>Parameter name</td>
<td>The parameter name that needs to be specified in API calls, e.g. in a parameter list for gx.analysis</td>
</tr>
<tr>
<td>Short description</td>
<td>Short description of the parameter</td>
</tr>
<tr>
<td>Type</td>
<td>Data type of the input, e.g. <code>data-element-path</code> expects a string which is a platform path of data element</td>
</tr>
<tr>
<td>Class</td>
<td>Java class of a data element, e.g. ru.biosoft.table.TableDataCollection expects a path to refer to a tabular data</td>
</tr>
<tr>
<td>Required</td>
<td>Indicates if a value must be specified</td>
</tr>
<tr>
<td>Description</td>
<td>Possibly longer description of the parameter</td>
</tr>
</tbody>
</table>
<h3 id="output-from-apps-including-parameters">Output from <em>apps</em> including parameters</h3>
<table>
<thead>
<tr>
<th>Tool folder/name</th>
<th>API name</th>
<th>Parameter name</th>
<th>Short description</th>
<th>Type</th>
<th>Class</th>
<th>Required</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Data manipulation/Annotate diagram</td>
<td>Annotate diagram</td>
<td>inputDiagram</td>
<td>Input diagram</td>
<td>data-element-path</td>
<td>biouml.model.Diagram</td>
<td>true</td>
<td>Diagram to annotate</td>
</tr>
<tr>
<td>Data manipulation/Annotate diagram</td>
<td>Annotate diagram</td>
<td>table</td>
<td>Annotation table</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>true</td>
<td>Table with diagram ids and annotations</td>
</tr>
<tr>
<td>Data manipulation/Annotate diagram</td>
<td>Annotate diagram</td>
<td>column</td>
<td>Annotation column</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>Column with annotations</td>
</tr>
<tr>
<td>Data manipulation/Annotate diagram</td>
<td>Annotate diagram</td>
<td>outputDiagram</td>
<td>Output diagram</td>
<td>data-element-path</td>
<td>biouml.model.Diagram</td>
<td>false</td>
<td>Path to store annotated diagram</td>
</tr>
<tr>
<td>Data manipulation/Annotate table</td>
<td>Annotate table</td>
<td>inputTablePath</td>
<td>Experiment</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>true</td>
<td>Table with experimental(test) data</td>
</tr>
<tr>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>miRNAs</td>
<td>miRNAs</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>true</td>
<td>miRNAs</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>targetGenes</td>
<td>targetGenes</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>true</td>
<td>targetGenes</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>siteModelCollection</td>
<td>siteModelCollection</td>
<td>data-element-path</td>
<td>ru.biosoft.bsa.SiteModelCollection</td>
<td>true</td>
<td>siteModelCollection</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>ensembl</td>
<td>ensembl</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>ensembl</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>scoreType</td>
<td>scoreType</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>scoreType</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>backgroundGenes</td>
<td>backgroundGenes</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>true</td>
<td>backgroundGenes</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>from</td>
<td>from</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>from</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>to</td>
<td>to</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>to</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>targetScanDB</td>
<td>targetScanDB</td>
<td>data-element-path</td>
<td>ru.biosoft.access.SqlDataCollection</td>
<td>true</td>
<td>targetScanDB</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>miRNAPromoterScoreThreshold</td>
<td>miRNAPromoterScoreThreshold</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>miRNAPromoterScoreThreshold</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>targetGenePromoterScoreThreshold</td>
<td>targetGenePromoterScoreThreshold</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>targetGenePromoterScoreThreshold</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>miRNATargetScoreThreshold</td>
<td>miRNATargetScoreThreshold</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>miRNATargetScoreThreshold</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>scoreThreshold</td>
<td>scoreThreshold</td>
<td>code-string</td>
<td></td>
<td>false</td>
<td>scoreThreshold</td>
</tr>
<tr>
<td>miRNA analysis/miRNA feed forward loops</td>
<td>miRNA feed forward loops</td>
<td>outTable</td>
<td>outTable</td>
<td>data-element-path</td>
<td>ru.biosoft.table.TableDataCollection</td>
<td>false</td>
<td>outTable</td>
</tr>
</tbody>
</table>
<h2 id="fetching-parameter-information-for-specific-tools-using-the-parameters-utility">Fetching parameter information for specific tools using the <em>parameters</em> utility</h2>
<p>The program <em>parameters</em> of the genexplain-api package extracts parameter
information for specified platform tools.</p>
<p>The following example JSON fetches parameter descriptions for <em>Annotate table</em> and
the Galaxy tool <em>variant_effect_predictor</em>. The <code>server</code>, <code>user</code>
and <code>password</code> properties need to be set with valid input values.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code><span class="p">{</span>
    <span class="nt">&quot;server&quot;</span><span class="p">:</span>          <span class="s2">&quot;https://platform.genexplain.com&quot;</span><span class="p">,</span>
    <span class="nt">&quot;user&quot;</span><span class="p">:</span>            <span class="s2">&quot;someuser@email.io&quot;</span><span class="p">,</span>
    <span class="nt">&quot;password&quot;</span><span class="p">:</span>        <span class="s2">&quot;12345&quot;</span><span class="p">,</span>
    <span class="nt">&quot;tools&quot;</span><span class="p">:</span>           <span class="p">[</span><span class="s2">&quot;Annotate table&quot;</span><span class="p">,</span> <span class="s2">&quot;variant_effect_predictor&quot;</span><span class="p">]</span>
<span class="p">}</span>
</code></pre></div>
</td></tr></table>

<p>The following shell invokes <em>parameters</em> with the JSON object above
stored in a file named <em>specific_tool_parameters.json</em>.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code>java -jar genexplain-api.jar parameters specific_tool_parameters.json
</code></pre></div>
</td></tr></table>

<p>The <a href="#output-produced-by-parameters-for-platform-tools">output</a> is printed to standard output (optionally, an additional JSON parameter <code>outfile</code> can
be provided to specify a local output file) and is (partially) shown below. It consists of a JSON object with
one property for each platform tool that was specified in the <em>tools</em> list.</p>
<p>The input options of a platform tool are contained in a JSON array, each represented by an object describing
the input option. The order of input parameters in this array is the same as in the graphical web interface.</p>
<p>The <code>displayName</code> property shows the parameter title that appears in the graphical web interface. <strong>However</strong>,
the parameter name that needs to be specified in an API call is the <code>name</code> property.
E.g. the first parameter of <em>Annotate table</em> with <code>displayName</code> <em>Experiment</em> has the <code>name</code> <em>inputTablePath</em>. The latter
needs to be used in function calls.</p>
<p>An input option object provides further valuable information such as the type of data that is expected. E.g.
the <em>Experiment</em> needs to be provided as a <code>type</code> <em>data-element-path</em>, which means a path for a
data element in the platform workspace and data element on that path must be a table (<code>elementClass</code> <em>...TableDataCollection</em>).</p>
<p>If an input option expects a value from a predefined list, the available values to select are also listed, see
e.g. output and description for <a href="first_example.html#output-for-the-functional-classification-tool">Functional classification</a>.</p>
<p>The <em>geneXplainR</em> method to fetch parameter information for platform tools is implemented in the
function <code>gx.analysis.parameters</code>. However, the <em>parameters</em> commandline tool provides more details
about each input option. Therefore, we recommend its use also for geneXplainR projects.</p>
<h3 id="output-produced-by-parameters-for-platform-tools">Output produced by <em>parameters</em> for platform tools</h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><code><span class="p">{</span>
    <span class="nt">&quot;Annotate table&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="p">{</span>
            <span class="nt">&quot;canBeNull&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;promptOverwrite&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;displayName&quot;</span><span class="p">:</span> <span class="s2">&quot;Experiment&quot;</span><span class="p">,</span>
            <span class="nt">&quot;name&quot;</span><span class="p">:</span> <span class="s2">&quot;inputTablePath&quot;</span><span class="p">,</span>
            <span class="nt">&quot;icon&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.table:ru/biosoft/table/resources/table.gif&quot;</span><span class="p">,</span>
            <span class="nt">&quot;description&quot;</span><span class="p">:</span> <span class="s2">&quot;Table with experimental(test) data&quot;</span><span class="p">,</span>
            <span class="nt">&quot;readOnly&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;elementClass&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.table.TableDataCollection&quot;</span><span class="p">,</span>
            <span class="nt">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;data-element-path&quot;</span><span class="p">,</span>
            <span class="nt">&quot;value&quot;</span><span class="p">:</span> <span class="s2">&quot;(select element)&quot;</span><span class="p">,</span>
            <span class="nt">&quot;elementMustExist&quot;</span><span class="p">:</span> <span class="kc">true</span><span class="p">,</span>
            <span class="nt">&quot;multiSelect&quot;</span><span class="p">:</span> <span class="kc">false</span>
        <span class="p">},</span>

        <span class="s2">&quot;...&quot;</span><span class="p">,</span>

        <span class="p">{</span>
            <span class="nt">&quot;displayName&quot;</span><span class="p">:</span> <span class="s2">&quot;Remove duplicate annotations&quot;</span><span class="p">,</span>
            <span class="nt">&quot;name&quot;</span><span class="p">:</span> <span class="s2">&quot;replaceDuplicates&quot;</span><span class="p">,</span>
            <span class="nt">&quot;description&quot;</span><span class="p">:</span> <span class="s2">&quot;If input table was already annotated by this source, old annotation columns will be removed from the result&quot;</span><span class="p">,</span>
            <span class="nt">&quot;readOnly&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;bool&quot;</span><span class="p">,</span>
            <span class="nt">&quot;value&quot;</span><span class="p">:</span> <span class="s2">&quot;true&quot;</span>
        <span class="p">},</span>
        <span class="p">{</span>
            <span class="nt">&quot;auto&quot;</span><span class="p">:</span> <span class="s2">&quot;on&quot;</span><span class="p">,</span>
            <span class="nt">&quot;displayName&quot;</span><span class="p">:</span> <span class="s2">&quot;Output table&quot;</span><span class="p">,</span>
            <span class="nt">&quot;icon&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.table:ru/biosoft/table/resources/table.gif&quot;</span><span class="p">,</span>
            <span class="nt">&quot;description&quot;</span><span class="p">:</span> <span class="s2">&quot;Output table&quot;</span><span class="p">,</span>
            <span class="nt">&quot;readOnly&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;data-element-path&quot;</span><span class="p">,</span>
            <span class="nt">&quot;canBeNull&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;promptOverwrite&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;name&quot;</span><span class="p">:</span> <span class="s2">&quot;outputTablePath&quot;</span><span class="p">,</span>
            <span class="nt">&quot;elementClass&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.table.TableDataCollection&quot;</span><span class="p">,</span>
            <span class="nt">&quot;value&quot;</span><span class="p">:</span> <span class="s2">&quot;(select element)&quot;</span><span class="p">,</span>
            <span class="nt">&quot;elementMustExist&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;multiSelect&quot;</span><span class="p">:</span> <span class="kc">false</span>
        <span class="p">}</span>
    <span class="p">],</span>
    <span class="nt">&quot;variant_effect_predictor&quot;</span><span class="p">:</span> <span class="p">[</span>
        <span class="p">{</span>
            <span class="nt">&quot;canBeNull&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;promptOverwrite&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;displayName&quot;</span><span class="p">:</span> <span class="s2">&quot;Input vcf file&quot;</span><span class="p">,</span>
            <span class="nt">&quot;name&quot;</span><span class="p">:</span> <span class="s2">&quot;input&quot;</span><span class="p">,</span>
            <span class="nt">&quot;icon&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.bsa:ru/biosoft/bsa/resources/track.gif&quot;</span><span class="p">,</span>
            <span class="nt">&quot;description&quot;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
            <span class="nt">&quot;readOnly&quot;</span><span class="p">:</span> <span class="kc">false</span><span class="p">,</span>
            <span class="nt">&quot;elementClass&quot;</span><span class="p">:</span> <span class="s2">&quot;ru.biosoft.bsa.Track&quot;</span><span class="p">,</span>
            <span class="nt">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;data-element-path&quot;</span><span class="p">,</span>
            <span class="nt">&quot;value&quot;</span><span class="p">:</span> <span class="s2">&quot;(select element)&quot;</span><span class="p">,</span>
            <span class="nt">&quot;elementMustExist&quot;</span><span class="p">:</span> <span class="kc">true</span><span class="p">,</span>
            <span class="nt">&quot;multiSelect&quot;</span><span class="p">:</span> <span class="kc">false</span>
        <span class="p">},</span>
        <span class="s2">&quot;...&quot;</span>
    <span class="p">]</span>
<span class="p">}</span>
</code></pre></div>
</td></tr></table>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="first_example.html" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                A first example&#58; Enrichment of gene functions in a gene set
              </div>
            </div>
          </a>
        
        
          <a href="api_overview.html" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Overview of APIs
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; 2022 geneXplain GmbH, Wolfenbuettel, Germany
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="assets/javascripts/vendor.77e55a48.min.js"></script>
      <script src="assets/javascripts/bundle.9554a270.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: ".",
          features: [],
          search: Object.assign({
            worker: "assets/javascripts/worker/search.4ac00218.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
    
  </body>
</html>